ValveResourceFormat/IO/MaterialExtract.cs

```diff
@@ -147,6 +149,15 @@ public enum Channel
         }
     };
 
+    public static readonly Dictionary<string, ChannelMapping[]> GltfTextureMappings = new()
+    {
+        ["BaseColor"] = new[] { (Channel.RGB, "TextureColor"), (Channel.A, "TextureTranslucency") },
+        ["MetallicRoughness"] = new[] { (Channel.G, "TextureRoughness"), (Channel.B, "TextureMetalness") },
```

Comment:
How does this play along with MRTextureGenerator?

Reply:
For the `MetallicRoughness` entry, if there happens to be a source2 shader that defines a texture with same exact inputs as above then the generator will be skipped (desired).

However there isn't such shader in source2 at this time, and a more realistic set of inputs would occupy the red channel too, as even the glTF spec recommends putting Occlusion/Emmisive in there (or in alpha). I might need to make it "pattern match" for such scenarios (source2 {R, G, B} -> gltf {G, B}). Otherwise it's just there for consistency.